packages/dashboard-frontend/src/containers/Loader/Factory/getGitRemotes.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2018-2021 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import common from '@eclipse-che/common';
+
+export interface GitRemote {
+  name: string;
+  url: string;
+}
+
+/**
+ * Returns parsed Git remotes given a string in one of the following
+ * formats:
+ *
+ * 1. Explicit name and URL
+ * {{origin,https://git...},{upstream,https://git...},...}
+ *
+ * 2. URLs only, name is implicit: first is origin, second is upstream, subsequent are fork1, fork2
+ * {https://git...,https://git...,...}
+ *
+ * @param remotes input string to parse
+ * @returns parsed array of Git remotes
+ */
+export function getGitRemotes(remotes: string): GitRemote[] {
+  if (!remotes || remotes.length === 0) {
+    return [];
+  }
+  const remotesArray = parseRemotes(remotes);
+  if (Array.isArray(remotesArray[0])) {
+    return parseNameAndUrls(remotesArray, remotes);
+  }
+  return parseUrls(remotesArray, remotes);
+}
+
+function parseRemotes(remotes: string): string[] {
+  try {
+    return JSON.parse(convertToArrayString(remotes));
+  } catch (e) {
+    throw `Unable to parse remotes. ${common.helpers.errors.getMessage(e)}`;
+  }
+}
+
+/**
+ * Replaces braces ({}) with square brackets ([]) and adds
+ * quotation marks (") around strings
+ * @param str
+ * @returns string representing an array
+ */
+function convertToArrayString(str: string): string {
+  return (
+    str
+      .replaceAll(' ', '')
+      .replaceAll('{', '[')
+      .replaceAll('}', ']')
+      /* eslint-disable no-useless-escape */
+      .replaceAll(/(\[([^\[]))/g, '["$2')
+      .replaceAll(/(([^\]])\])/g, '$2"]')
+      .replaceAll(/(([^\]]),([^\[]))/g, '$2","$3')
```

Comment:
Consider including short comments to make parsing these ones a little easier (especially with the capture groups), e.g. `replace '[<string>' with '["<string>`

Reply:
Thank you, fixed